### ISSUE TITLE
Where are the saved items stored?

### ISSUE THREAD

**Issue by ifnullzero** [alexpdraper/reading-list#135]
I was using this add on but after the last FF update it seems to have stopped working - I can still see the icon in the URL bar but clicking it does nothing. It seems "dead". I'm just wondering where this add on stores items that I have previously saved.

I saved about a dozen links and now I cannot access them. Is there any way to access them now? In some local storage somewhere? If so how would I find them? Thanks.

**Comment by alexpdraper** [comment]
Hey there, the reading list items are stored in your local storage unless you are logged in, and then they’ll be synced with your account. They’re accessible through the [browser extension storage API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync). You can see the raw storage output by navigating directly to the popup page for the reading list in a Firefox window. I believe the URL is `moz-extension://860f3270-6124-5644-9657-d5387eea5eae/popup.html`. Then you can open your developer tools by going to `Tools` > `Browser Tools` > `Web Developer Tools`.

<img width="940" alt="Screenshot of the menu to open the “Web Developer Tools” in Firefox" src="https://user-images.githubusercontent.com/6441503/123465494-0c703d80-d5a3-11eb-8f06-b3e316ab55e0.png">

In the developer tools panel, select `Console`. Add this code and hit enter to see all the items that are stored for your Reading List.

~~~javascript
browser.storage.sync.get(null, (items) => {
  console.log(items);
});
~~~

Hope this helps!

**Comment by alexpdraper** [comment]
You can also download all the items stored in your Reading List by going to the Reading List settings and clicking `Export`.

<img width="701" alt="Screenshot of the Reading List settings page" src="https://user-images.githubusercontent.com/6441503/123466407-278f7d00-d5a4-11eb-9349-e9ca3328b337.png">

**Comment by ifnullzero** [comment]
Thank you for responding, I was able to get them via the Export button method. 

About the other method (moz-extension URL). I'm interested to see how that works. I tried the URL you mentioned (moz-extension://860f3270-6124-5644-9657-d5387eea5eae/popup.html) but it did not do anything -- when I try it the page just remains where it currently is. Is the URL correct? How would I find out what the correct URL is? Thx.

**Comment by alexpdraper** [comment]
The ID for the reading list might be different for you. I opened the reading list popup, right-clicked inside of it, and selected `View Page Source`. Then I removed the `view-source:` from the beginning of the URL that gets opened. Hopefully that works 🤞.

**Comment by formlessdao** [comment]
Can the addon import large amounts of links? I have a 50000 links list I exported from Onetab but I can't seem to import it in Reading List.

**Comment by alexpdraper** [comment]
> Can the addon import large amounts of links? I have a 50000 links list I exported from Onetab but I can't seem to import it in Reading List.

It cannot. There are limits imposed by the browser on how much storage space the extension can use. You’ll probably get capped at 400–500 items.

**Comment by formlessdao** [comment]
So reading list can not contain more than max 500 items even if i add them one by one using Reading List itself?

**Comment by alexpdraper** [comment]
Yes. It’s not a limit of the import/export functionality, it’s a restriction on storage limits set by the browser. Not much we can do about it. The max items will probably be less than 500. It depends on the length of the URLs/titles in the individual reading list items added.

**Comment by istvandbsd** [comment]
Hi, for me it capped at 250 items. Unfortunately it shows as if it could save more, but then they aren't getting into the list. (Seemed like a bug, now it seems more like a UX issue, but I was too lazy to open an issue 😅 )
